### PR TITLE
Create health daemon for monitoring sbp messages

### PR DESCRIPTION
**Pull request by benjaminaltieri**
New package folder with health_daemon resources
have been created to allow for generic monitoring
of sbp message content for the purpose of logging
information to the console via sbp log. The newly
created health_thread module allows you to register
a message callback and a timer callback while
managing both internally using zloop timer tickets.

The first thread is a simple message content check
of the baseline ecef message but illustrates the
proper usage and setup of a health thread module.

### CODE REVIEW

**Comment by silverjam** [comment]
I'm confused, why are these called threads?  They don't seem to actually use pthread_create, but just register callbacks with zmq?

**Comment by benjaminaltieri** [comment]
I can replace the nomenclature if it's confusing. The idea is that each module will run with it's own data that is shared between timer callbacks and message event callbacks, with some shared context provided by the health_daemon.

**Comment by silverjam** [comment]
I would just change "thread" to something else, like "check", or something else.  Otherwise looks fine, I'm assuming all the checks work?

File: package/health_daemon/src/baseline_thread.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+{
+  (void)thread;
+  (void)sender_id;
+  (void)len;
+  log_fn_t log_fn = (log_fn_t)ctx;
+  msg_baseline_ecef_t *msg = (void*)msg_;
+  float x = msg->x, y = msg->y, z = msg->z;
+  float distance = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));
+  if (distance > BASELINE_THRESHOLD) {
+    log_fn(LOG_WARNING, "Baseline Distance Over Threshold: %fm", distance);
```
Comment (silverjam) [inline package/health_daemon/src/baseline_thread.c]: How often is this warning going to be issued?  We may want to rate limit to every 5 or 10s?

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2018010520-23-gff890ea
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-23-gff890ea/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2018010520-23-gff890ea/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-23-gff890ea

**Comment by benjaminaltieri** [comment]
Naming refactored. Merging now so this can get into 1.4

**Comment by anonymous-travis-swiftnav** [comment]
## v1.2.0-develop-2018010520-31-g3331a32
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-31-g3331a32/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.2.0-develop-2018010520-31-g3331a32/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.2.0-develop-2018010520-31-g3331a32

**Pull request merged**